covid19-dashboard/dc-covid19/src/App.tsx

```diff
@@ -14,186 +14,122 @@
-                                                                  panelId={panelId}
-                                                                  datePicked={this.state.datePicked}
-                                                                  region={this.state.selectedRegion}
-                                                                  ref_={this.refs_[panelId]}
-                                                                  selectedShowTopN={this.state.selectedShowTopN}/>)
+/**
+ * Returns an array containing [initialDate, lastDate]
+ * For example, if we cared about the data from 01-01-10 to 1 week ago.
+ * initialDate = 2020-01-10, deltaDays = 7, returns [2020-01-03, 2020-01-10]
+ * @param ISOdate: date to look at
+ * @param deltaDays: number of days before our ISOdate to look at
+ */
+const getRangeOfDates = (
+  ISOdate: string,
+  deltaDays: number
+): [string, string] => {
+  // [ISOdate - deltaDays, ISOdate]
+  return [addOrSubtractNDaysToDate(ISOdate, -deltaDays), ISOdate];
+};
 
-        return (
-            <div className={"container"}>
-                <SideNav handleScrollOnRef={this.handleScrollOnRef}/>
-                <div className={"main-content"}>
-                    <h1 className={"main-title"}>
-                        {Configuration.TITLE + ' '}
-                        <span style={{color: "#990001"}}>{Configuration.SUBTITLE}</span>
-                    </h1>
-                    <OptionPanel handleSelectUpdate={this.handleSelectUpdate}
-                                 geoIdToName={this.state.allData[3]}
-                                 defaultShowTopN={this.state.selectedShowTopN}
-                                 datesToPick={this.state.datesToPick}
-                                 datePicked={this.state.datePicked}
-                                 availableRegions={this.state.availableRegions}
-                                 selectedRegion={this.state.selectedRegion}/>
-                        {rows}
-                </div>
-                <h5 className={"footer"}>{Configuration.FOOTER}</h5>
-            </div>
-        )
-    }
-}
+/**
+ * Given a map of geoId->[name, belongToRegion], it returns a list of the geoId's that belong to the belongToRegion
+ * @param geoIdToType: geoId->[name, belongToRegion]
+ * @param belongToRegion: can be any geoId
+ */
+const filterGeoIdThatBelongTo = (
+  geoIdToType: {geoId: string[]} | {},
+  belongToRegion: string
+): string[] => {
+  if (belongToRegion === 'County')
+    return Object.keys(geoIdToType).filter(
+      geoId => geoIdToType[geoId][1] !== 'country/USA'
+    );
+  else if (belongToRegion === 'State')
+    return Object.keys(geoIdToType).filter(
+      geoId => geoIdToType[geoId][1] === 'country/USA'
+    );
+  else
+    return Object.keys(geoIdToType).filter(
+      geoId => geoIdToType[geoId][1] === belongToRegion
+    );
+};
 
-export default App;
+/**
```

Comment:
try looking for in lodash library, there should be something doing this

Reply:
Done. Take a look and let me know if this is what you meant.
I also changed the algorithm of the function. Now it handles multiple countries.

Regions are stored as such:  **{geoId: [name, belongsTo]}**

1. Get all **States** (belongsTo === 'country/USA').
2. Get all Counties (belongsTo in set **States**)
4. Get all Counties that belong to a State (belongsTo === geoId/XYZ)